pkg/config/store/store2.go

```diff
@@ -62,8 +62,31 @@ type BackEndResource struct {
 
 // Resource represents a resources with converted spec.
 type Resource struct {
-	Metadata ResourceMeta
-	Spec     proto.Message
+	Metadata ResourceMeta  `json:"metadata"`
+	Spec     proto.Message `json:"spec"`
+	Status   Status        `json:"status"`
+}
+
+// Code is textual status code in Status.
+type Code string
+
+const (
+	// Active - the resource is active.
+	Active Code = "active"
+	// PendingUpdate - the resource is pending updation.
+	PendingUpdate Code = "pendingUpdate"
+	// PendingDelete - the resource is pending deletion.
+	PendingDelete Code = "pendingDelete"
+	// InError - the resource is in error.
+	InError Code = "error"
+)
+
+// Status is the current status of a resource.
+type Status struct {
+	// Code for this status.
+	Code Code `json:"code"`
```

Comment:
this is potentially confusing, with status.code on grpc reflecting a few well-known status codes (which are different than just strings).

Reply:
Changed names to be like `PodStatus`